typegraph/node/sdk/src/tg_deploy.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+import { TypegraphOutput } from "./typegraph.js";
+import { wit_utils } from "./wit.js";
+
+export interface BasicAuth {
+  username: string;
+  password: string;
+}
+
+export interface TypegraphDeployParams {
+  baseUrl: string;
+  auth?: BasicAuth;
+  secrets: Record<string, string>;
+  cliVersion: string;
+}
+
+export async function tgDeploy(
+  typegraph: TypegraphOutput,
+  params: TypegraphDeployParams,
+) {
+  const { baseUrl, cliVersion, secrets, auth } = params;
+  const { serialized } = typegraph;
+
+  const headers = new Headers();
+  headers.append("Content-Type", "application/json");
+  if (auth) {
```

Comment:
Why not defaults to the credentials provided in the `metatype.yaml`? We could pass them through environment variables...

Same for the typegate url and secrets...

Reply:
My assumption was that `metatype.yaml` is for a better dx with the cli, we could add support for that though, but I am not sure if the env variable approach is the proper way to do it in this context, we can instead manually load the yml file from the typegraph and use its content in memory.